docs/rancher.md

```diff
@@ -0,0 +1,23 @@
+# Deploying Antrea on Rancher Managed Cluster
+
+<!-- toc -->
+- [Deploy Rancher Server](#deploy-rancher-server)
+- [Deploy Antrea Using Rancher UI](#deploy-antrea-using-rancher-ui)
+<!-- /toc -->
+
+## Deploy Rancher Server
+
```

Comment:
We should add version information with which rancher and antrea was tried to remove ambiguity, there might be many queries raised if the previous version did not work.